packages/rooks/src/hooks/useMediaMatch.ts

```diff
@@ -11,13 +11,20 @@ import { useEffect, useMemo, useState } from "react";
  * @see https://rooks.vercel.app/docs/useMediaMatch
  */
 function useMediaMatch(query: string): boolean {
-  const matchMedia = useMemo<MediaQueryList>(
-    () => window.matchMedia(query),
-    [query]
+  const isWindowDefined = typeof window !== "undefined" && window.matchMedia;
+  const matchMedia = useMemo<MediaQueryList | undefined>(
+    () => (isWindowDefined ? window.matchMedia(query) : undefined),
+    [query, isWindowDefined]
+  );
+  const [matches, setMatches] = useState<boolean>(() =>
+    (isWindowDefined && matchMedia ? matchMedia.matches : false)
   );
-  const [matches, setMatches] = useState<boolean>(() => matchMedia.matches);
 
   useEffect(() => {
+    if (!isWindowDefined || !matchMedia) {
```

Comment:
I wonder if this is needed, since effects only run in the dom. Unless I am missing something.

Reply:
When I remove that, the test no longer passes, and I'm not sure how to fix it. Here's the output:

```
rooks:test: Summary of all failing tests
rooks:test: FAIL __tests__/useMediaMatch.spec.tsx
rooks:test:   ● useMediaMatch › should track a boolean
rooks:test:
rooks:test:     expect(received).toBe(expected) // Object.is equality
rooks:test:
rooks:test:     Expected: true
rooks:test:     Received: false
rooks:test:
rooks:test:       51 |
rooks:test:       52 |     // Invoking the listener changes the value
rooks:test:     > 53 |     const l =
rooks:test:          |              ^
rooks:test:       54 |       expectDefined<(event_: MediaQueryListEventMap["change"]) => void>(
rooks:test:       55 |         listener
rooks:test:       56 |       );
rooks:test:
rooks:test:       at Object.<anonymous> (__tests__/useMediaMatch.spec.tsx:53:28)
```

And the relevant code:

```
  expect(result.current).toBe(true);

  // Invoking the listener changes the value
  const l =
    expectDefined<(event_: MediaQueryListEventMap["change"]) => void>(
      listener
    );
  act(() => l({ matches: false } as any));
  expect(result.current).toBe(false);
```

When it runs `l()` near the end of this snippet, since `window` is not defined, it returns the previous value of `isMatches` which was `true`.